h/views/api_users.py

```diff
@@ -89,6 +90,8 @@ def _request_client(request):
         raise ClientUnauthorized()
     if client.secret is None:  # client is not confidential
         raise ClientUnauthorized()
+    if client.grant_type != GrantType.client_credentials:  # client not allowed to create users
```

Comment:
This feels to me like it’s giving the views quite a lot of knowledge about the internals of the `AuthClient` model. The fact that this has comments to explain the intention suggests that there’s maybe a concept in here that needs a name and doesn’t have one. Would it be worth pulling out the logic of which clients are allowed to create users under their authority, either as a property on the model¹ or on its own object?

¹ I don’t know how you specifically – or we collectively – feel about adding extra logic onto the models, but we do have the [`Group.is_public`](https://github.com/hypothesis/h/blob/master@%7B2017-08-04%7D/h/models/group.py#L103-L105) property as one example where we’ve taken a model concept and named it.